core/shared/src/main/scala-3/cats/effect/cps.scala

```diff
@@ -31,10 +32,9 @@ object cps {
     transparent inline def await: A = _root_.cps.await[F, A, F](self)
   }
 
-  implicit def catsEffectCpsMonadPureMemoization[F[_]](implicit F: Concurrent[F]): CpsMonadMemoization.Pure[F] =
```

Comment:
Does memoization has to do with automatic colouring in the context of dotty-cps? If not, any reason not to keep it ? 

MiMa is complaining about bincompat, so it's either a case of keeping it, or prepping the [version](https://github.com/typelevel/cats-effect-cps/blob/main/build.sbt#L19) so that the next release is 0.5.0

Reply:
In the early days, memoization was needed in some internal interfaces for async but actually called only during automatic coloring.
Now - dotty-cps-async requires a memoization interface only if automatic coloring is enabled.
From another side, we can restore memoization for compatibility, overhead is small.